packages/ipfs-daemon/src/create-repo.ts

```diff
@@ -0,0 +1,79 @@
+import S3 from 'aws-sdk/clients/s3';
+import IPFSRepo from 'ipfs-repo';
+import S3Store from 'datastore-s3'
+
+// A mock lock
+const notALock = {
+  getLockfilePath: () => {
+    // Do Nothing
+  },
+  lock: () => notALock.getCloser(),
+  getCloser: () => ({
+    close: () => {
+      // Do Nothing
+    },
+  }),
+  locked: () => false,
+};
+
+/**
+ * A convenience method for creating an S3 backed IPFS repo
+ *
+ * @param {Object} S3Store
+ * @param {Object} options
+ * @param {Object} s3Options
+ * @returns {Object}
+ */
+export function createRepo(options: any, s3Options: any): any {
+  const { bucket, region, accessKeyId, secretAccessKey } = s3Options;
+
+  const { path, createIfMissing, lock } = options;
+
+  const storeConfig = {
+    s3: new S3({
+      params: {
+        Bucket: bucket,
+      },
+      region,
+      accessKeyId,
+      secretAccessKey,
+    }),
+    createIfMissing,
+  };
+
+  // If no lock is given, create a mock lock
+  const effectiveLock = lock || notALock;
+
+  return new IPFSRepo(path, {
+    storageBackends: {
+      root: S3Store,
+      blocks: S3Store,
+      keys: S3Store,
+      datastore: S3Store,
+      pins: S3Store,
+    },
+    storageBackendOptions: {
+      root: {
+        ...storeConfig,
+        sharding: false,
+      },
+      blocks: {
+        ...storeConfig,
+        sharding: true,
+      },
+      keys: {
+        ...storeConfig,
+        sharding: false,
+      },
+      datastore: {
+        ...storeConfig,
+        sharding: false,
+      },
+      pins: {
+        ...storeConfig,
+        sharding: true,
```

Comment:
Not positive this was previously sharded

Reply:
It was not.